loopin.go

```diff
@@ -694,19 +694,23 @@ func (s *loopInSwap) publishTimeoutTx(ctx context.Context,
 // update notification.
 func (s *loopInSwap) persistAndAnnounceState(ctx context.Context) error {
 	// Update state in store.
-	err := s.store.UpdateLoopIn(
+	if err := s.persistState(); err != nil {
+		return err
+	}
+
+	// Send out swap update
+	return s.sendUpdate(ctx)
+}
+
+// persistState updates the swap state on disk.
+func (s *loopInSwap) persistState() error {
```

Comment:
nit: I'd probably merge this commit and the previous one.

Reply:
I thought this would make it easier to review, especially because there is no rewriting of prior changes

Reply:
Does this need to be split out at all if we always call `persistAndAnnounce`?

Reply:
Another vote for making state updates cleaner :)

Reply:
@carlaKC in this PR, `persistState` is also called without announce.